Shopipy.Web/Services/UserService.cs

```diff
@@ -0,0 +1,26 @@
+using Shopipy.UserManagement.Dtos;
+using System.Net.Http.Headers;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Shopipy.Web.Services;
+
+public class UserService(IHttpClientFactory httpClientFactory)
+{
+    private readonly HttpClient _httpClient = httpClientFactory.CreateClient("Shopipy.ApiService");
+
+    public async Task<UserResponseDto?> GetUserByIdAsync(string userId, string token)
+    {
+        _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+
+        var response = await _httpClient.GetAsync($"/Users/{userId}");
+        if (!response.IsSuccessStatusCode) return null;
+
+        var content = await response.Content.ReadAsStringAsync();
+
+        var options = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
+        options.Converters.Add(new JsonStringEnumConverter());
+
+        return JsonSerializer.Deserialize<UserResponseDto>(content, options);
```

Comment:
These lines will be the same for most of the api request, you should abstract it to avoid repetition. I would suggest creating generic HttpService